### DISCUSSION TITLE
Changing the middleware to 'auth' or to some role redirects to homepage?

### DISCUSSION THREAD

**Discussion by milanchheda-cactus** [lorisleiva/artisan-ui#5] [Q&A] [answered]
Hey, 

First of all, excellent package. Loved it. Thanks a lot.

I am trying to change the middleware in the config file from 
```
'middleware' => ['web', AuthorizeArtisanUI::class]
```
to
```
'middleware' => ['auth', AuthorizeArtisanUI::class]
``` 

After the change, when I login and try to access `/artisan` route, its redirecting me homepage. Can you tell me what am I missing here?

We are also using Spatie/laravel-permission package and basically we want to restrict `/artisan` route to certain role.

**Comment by lorisleiva** [comment, marked as answer]
Hi there 👋 

Is there any particular reasons why you're not using the `web` middleware group? Artisan UI relies on sessions to authenticate you to that page so by not using the `web` middleware group, I'm not sure that these would be set properly.

**Reply by milanchheda-cactus** [reply]
Hey, thanks for the response.

I did tried with `web` and it sure works but then how do we restrict this route `/artisan` it to certain roles?

**Reply by lorisleiva** [reply]
You can use the `ArtisanUI::auth` method for this. For example, it could look something like that:


```php
ArtisanUI::auth(function ($request) {
    // Allow everyone locally.
    if (app()->environment('local')) {
        return true;
    }

    // Otherwise, denied unauthenticated users.
    if (! $request->check()) {
        return false;
    }

    // Finally, ensure they have the right permission.
    return $request->user()->permissions()->where('name', 'artisan-ui')->exists();
});
```

**Reply by milanchheda-cactus** [reply]
Ok, I will try that and update this discussion. May I know where to write this method `ArtisanUI::auth(function ($request) {
`?

**Reply by lorisleiva** [reply]
You can write that in the `boot` method of any of your Service Providers. Usually, that would be in your `AppServiceProvider`.
